src/ng/templateRequest.js

```diff
@@ -68,8 +68,14 @@ function $TemplateRequestProvider() {
       // are included in there. This also makes Angular accept any script
       // directive, no matter its name. However, we still need to unwrap trusted
       // types.
-      if (!isString(tpl) || !$templateCache.get(tpl)) {
```

Comment:
Couldn't you just change this line to `... || isUndefined($templateCache.get(tpl))...` and leave everythng else unchanged ?